cli/commands.go

```diff
@@ -83,11 +83,26 @@ func commonInit(config *conf.MenderConfig, opts *runOptionsType) (*app.MenderPie
 		dirstore *store.DirStore
 	)
 	dirstore = store.NewDirStore(opts.dataStore)
+	var privateKey string
+	var sslEngine string
+	var static bool
+
 	if config.HttpsClient.Key != "" {
-		ks = store.NewKeystore(dirstore, config.HttpsClient.Key, config.HttpsClient.SSLEngine, true)
-	} else {
-		ks = store.NewKeystore(dirstore, conf.DefaultKeyFile, config.HttpsClient.SSLEngine, false)
+		privateKey = config.HttpsClient.Key
+		sslEngine = config.HttpsClient.SSLEngine
+		static = true
+	}
+	if config.Security.PKIPrivateKey != "" {
+		privateKey = config.Security.PKIPrivateKey
+		sslEngine = config.Security.SSLEngine
+		static = true
+	}
```

Comment:
I think we should print a warning if both are set. Otherwise the loading order makes sense to me.

Reply:
thank you Sir. [yes](https://github.com/mendersoftware/mender/pull/617/commits/f05da435e82f47f9a612d88c4024c91465869ea6#diff-bc0c8eb032e24a2f22bcc5cfe94b63abR168) Sir!